### PR TITLE
embed or-filter into IndexScan

### PR DESCRIPTION
**Pull request by czpmango**
Fixes [https://github.com/vesoft-inc/nebula-graph/issues/948](https://github.com/vesoft-inc/nebula-graph/issues/948)

### CODE REVIEW

File: src/planner/match/LabelIndexSeek.cpp [hunk 1]
```diff
@@ -85,6 +85,53 @@ StatusOr<SubPlan> LabelIndexSeek::transformNode(NodeContext* nodeCtx) {
...
+                auto& filterItems = static_cast<LogicalExpression*>(flattenFilter)->operands();
+                auto canBeEmbeded = [](Expression::Kind k) -> bool {
+                    return k == Expression::Kind::kRelEQ || k == Expression::Kind::kRelLT ||
+                           k == Expression::Kind::kRelLE || k == Expression::Kind::kRelGT ||
+                           k == Expression::Kind::kRelGE;
+                };
+                bool canBeEmbeded2IndexScan = true;
+                for (auto& f : filterItems) {
+                    if (!canBeEmbeded(f->kind())) {
+                        canBeEmbeded2IndexScan = false;
```
Comment (yixinglu) [inline src/planner/match/LabelIndexSeek.cpp:117]: break;

File: src/util/ExpressionUtils.cpp [hunk 2]
```diff
@@ -106,6 +106,38 @@ ExpressionUtils::pullOrsImpl(LogicalExpression *expr,
...
+        return e->clone().release();
+    };
+
+    return RewriteVisitor::transform(expr, std::move(matcher), std::move(rewriter));
+}
+
+Expression *ExpressionUtils::flattenInnerLogicalExpr(const Expression *expr) {
+    auto* andFlattenExpr = flattenInnerLogicalAndExpr(expr);
+    auto* allFlattenExpr = flattenInnerLogicalOrExpr(andFlattenExpr);
+    delete(andFlattenExpr);
```
Comment (yixinglu) [inline src/util/ExpressionUtils.cpp:270]: Use std::unique_ptr<Expression> to replace `delete`

File: src/util/ExpressionUtils.h [hunk 3]
```diff
@@ -101,6 +101,10 @@ class ExpressionUtils {
         Expression::Kind kind,
         const std::vector<std::unique_ptr<Expression>>& rels);
 
+    static Expression* flattenInnerLogicalAndExpr(const Expression* expr);
+    static Expression* flattenInnerLogicalOrExpr(const Expression* expr);
+    static Expression* flattenInnerLogicalExpr(const Expression* expr);
```
Comment (yixinglu) [inline src/util/ExpressionUtils.h:106]: Return std::unique_ptr;

Don't let user of these functions free the returned memory.

File: src/planner/match/LabelIndexSeek.cpp (hunk 1 above)
Comment (czpmango) [reply src/planner/match/LabelIndexSeek.cpp:117]: ACK.

File: src/util/ExpressionUtils.cpp (hunk 2 above)
Comment (czpmango) [reply src/util/ExpressionUtils.cpp:270]: ACK.

File: src/util/ExpressionUtils.h (hunk 3 above)
Comment (czpmango) [reply src/util/ExpressionUtils.h:106]: ACK.

**Review by yixinglu — APPROVED** [review]

**Review by jievince — APPROVED** [review]

**Pull request merged**